services/ux-backend/handlers/rotatekeys/handler.go

```diff
@@ -0,0 +1,89 @@
+package rotatekeys
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	contentTypeTextPlain                     = "text/plain"
+	onboardingValidationPublicKeySecretName  = "onboarding-ticket-key"
+	onboardingValidationPrivateKeySecretName = "onboarding-private-key"
+)
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, cl client.Client, namespace string) {
+	switch r.Method {
+	case "POST":
+		handlePost(r.Context(), w, cl, namespace)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+func handlePost(ctx context.Context, w http.ResponseWriter, cl client.Client, namespace string) {
+
+	w.Header().Set("Content-Type", contentTypeTextPlain)
+
+	privateKeySecret := &corev1.Secret{}
+	privateKeySecret.Name = onboardingValidationPrivateKeySecretName
+	privateKeySecret.Namespace = namespace
+	err := cl.Get(ctx, client.ObjectKeyFromObject(privateKeySecret), privateKeySecret)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			klog.Info("onboarding private key secret does not exist")
+			w.WriteHeader(http.StatusBadRequest)
+
+			// we cannot rotate keys due to any of:
+			// 1. we are not running in provider mode
+			// 2. keys are not being managed by us via onboarding job
```

Comment:
We should allow the rotation of keys even if we do not manage them. 
That is the automated way to move from customer-owned keys to ODF-owned keys.

Reply:
done.